#!/usr/bin/env python2
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()
gen.add("car_position_x", double_t, 1, "", 0.0, -10.0, 10.0)
gen.add("car_position_y", double_t, 1, "", 0.0, -0.0, 0.0) # deviation from 0 not factored into target point calculation!
gen.add("section_length", int_t, 1, "", 3, 1, 10)
gen.add("section_stride", int_t, 1, "", 3, 1, 10)
gen.add("segment_max_len", double_t, 1, "", 5.0, 0.0, 10.0)
gen.add("seg_seg_min_angle", int_t, 1, "", 105, 0, 180)
gen.add("same_type_limit", int_t, 1, "", 1, 0, 10)
gen.add("direction_bias", int_t, 1, "", 0, 0, 3)
gen.add("finish_line_trigger_radius", double_t, 1, "", 4.0, 0.0, 10.0)
gen.add("finish_line_trigger_corridor", double_t, 1, "", 1.5, 0.0, 10.0)
gen.add("target_distance_min", double_t, 1, "", 3.0, 0.0, 20.0)
gen.add("target_distance_max", double_t, 1, "", 10.0, 0.0, 20.0)
gen.add("target_distance_falloff", double_t, 1, "", 4.0, 0.0, 10.0)
gen.add("target_distance_scaler", double_t, 1, "", 2.0, 0.0, 10.0)
exit(gen.generate('bumblebob_pathfinder', "pathfinder", "Pathfinder"))
